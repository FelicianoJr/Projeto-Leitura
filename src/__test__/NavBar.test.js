import React from "react";
import { shallow } from "enzyme";
import Navbar from "../components/NavBar";
import { NavLink } from "react-router-dom";

const setup = (categories = [], disabled) => {
  const actions = {
    newPost: jest.fn(),
    sortPost: jest.fn()
  };

  const component = shallow(
    <Navbar {...actions} categories={categories} disabled={disabled} />
  );

  return {
    component: component,
    actions: actions,
    button: component.find("button"),
    select: component.find("select"),
    option: component.find("option"),
    navLink: component.find(NavLink)
  };
};

describe("<NavBar/>", () => {
  let categories;
  beforeEach(() => {
    categories = [
      {
        name: "react",
        path: "react"
      },
      {
        name: "redux",
        path: "redux"
      }
    ];
  });

  it("should render equal text <option>", () => {
    const { select } = setup();
    const option = select.find("option");
    expect(option.at(0).text()).toEqual("Maior Pontuação");
    expect(option.at(1).text()).toEqual("Menor Pontuação");
    expect(option.at(2).text()).toEqual("Postagens Recentes");
    expect(option.at(3).text()).toEqual("Postagens Antigas");
  });

  it("should call in select sortPost", () => {
    const { actions, select } = setup();
    select.simulate("change", { target: { value: "100" } });
    expect(actions.sortPost).toBeCalled();
  });

  it("should rende NavLink", () => {
    const { navLink } = setup(categories);
    expect(navLink.at(2).props().to).toEqual("/react");
    expect(navLink.at(3).props().to).toEqual("/redux");
  });

  it("should click newPost", () => {
    const { button, actions } = setup();
    button.simulate("click");
    expect(actions.newPost).toBeCalled();
  });

  it("shold component snapshot", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
//example change value
// const wrapper = shallow(<Login />);
//    wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}});

//    expect(wrapper.state('password')).toEqual('cats');
//   })
