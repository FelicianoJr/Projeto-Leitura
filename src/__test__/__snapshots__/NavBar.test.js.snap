// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> shold component snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    categories={Array []}
    disabled={undefined}
    newPost={[Function]}
    sortPost={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="navbar-brand"
          to="/"
>
          Udacity
</NavLink>,
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
>
          <ul
                    className="navbar-nav mr-auto"
          >
                    <li
                              className="nav-item active"
                    >
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        className="nav-link"
                                        to="/"
                              >
                                        Home 
                                        <span
                                                  className="sr-only"
                                        >
                                                  (current)
                                        </span>
                              </NavLink>
                    </li>
          </ul>
</div>,
        <form
          className="form-inline "
>
          <div
                    className="mr-sm-2"
          >
                    <select
                              className="form-control form-control-sm"
                              disabled={undefined}
                              id="sort"
                              onChange={[Function]}
                              value={undefined}
                    >
                              <option
                                        value="SCORE_BIGGER"
                              >
                                        Maior Pontuação
                              </option>
                              <option
                                        value="SCORE_SMALLER"
                              >
                                        Menor Pontuação
                              </option>
                              <option
                                        value="RECENT_POST"
                              >
                                        Postagens Recentes
                              </option>
                              <option
                                        value="OLD_POST"
                              >
                                        Postagens Antigas
                              </option>
                    </select>
          </div>
          <button
                    className="btn btn-outline-info btn-sm"
                    disabled={undefined}
                    onClick={[Function]}
                    type="button"
          >
                    Nova Postagem
          </button>
</form>,
      ],
      "className": "navbar navbar-expand-lg navbar-dark bg-dark justify-content-between",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Udacity",
          "className": "navbar-brand",
          "to": "/",
        },
        "ref": null,
        "rendered": "Udacity",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="navbar-nav mr-auto"
>
            <li
                        className="nav-item active"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav-link"
                                    to="/"
                        >
                                    Home 
                                    <span
                                                className="sr-only"
                                    >
                                                (current)
                                    </span>
                        </NavLink>
            </li>
</ul>,
          "className": "collapse navbar-collapse",
          "id": "navbarSupportedContent",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item active"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="nav-link"
                                to="/"
                >
                                Home 
                                <span
                                                className="sr-only"
                                >
                                                (current)
                                </span>
                </NavLink>
</li>,
              Array [],
            ],
            "className": "navbar-nav mr-auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-link"
                  to="/"
>
                  Home 
                  <span
                                    className="sr-only"
                  >
                                    (current)
                  </span>
</NavLink>,
                "className": "nav-item active",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": Array [
                    "Home ",
                    <span
                      className="sr-only"
>
                      (current)
</span>,
                  ],
                  "className": "nav-link",
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  "Home ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "(current)",
                      "className": "sr-only",
                    },
                    "ref": null,
                    "rendered": "(current)",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mr-sm-2"
>
              <select
                            className="form-control form-control-sm"
                            disabled={undefined}
                            id="sort"
                            onChange={[Function]}
                            value={undefined}
              >
                            <option
                                          value="SCORE_BIGGER"
                            >
                                          Maior Pontuação
                            </option>
                            <option
                                          value="SCORE_SMALLER"
                            >
                                          Menor Pontuação
                            </option>
                            <option
                                          value="RECENT_POST"
                            >
                                          Postagens Recentes
                            </option>
                            <option
                                          value="OLD_POST"
                            >
                                          Postagens Antigas
                            </option>
              </select>
</div>,
            <button
              className="btn btn-outline-info btn-sm"
              disabled={undefined}
              onClick={[Function]}
              type="button"
>
              Nova Postagem
</button>,
          ],
          "className": "form-inline ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="form-control form-control-sm"
                disabled={undefined}
                id="sort"
                onChange={[Function]}
                value={undefined}
>
                <option
                                value="SCORE_BIGGER"
                >
                                Maior Pontuação
                </option>
                <option
                                value="SCORE_SMALLER"
                >
                                Menor Pontuação
                </option>
                <option
                                value="RECENT_POST"
                >
                                Postagens Recentes
                </option>
                <option
                                value="OLD_POST"
                >
                                Postagens Antigas
                </option>
</select>,
              "className": "mr-sm-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="SCORE_BIGGER"
>
                    Maior Pontuação
</option>,
                  <option
                    value="SCORE_SMALLER"
>
                    Menor Pontuação
</option>,
                  <option
                    value="RECENT_POST"
>
                    Postagens Recentes
</option>,
                  <option
                    value="OLD_POST"
>
                    Postagens Antigas
</option>,
                ],
                "className": "form-control form-control-sm",
                "disabled": undefined,
                "id": "sort",
                "onChange": [Function],
                "value": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Maior Pontuação",
                    "value": "SCORE_BIGGER",
                  },
                  "ref": null,
                  "rendered": "Maior Pontuação",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Menor Pontuação",
                    "value": "SCORE_SMALLER",
                  },
                  "ref": null,
                  "rendered": "Menor Pontuação",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Postagens Recentes",
                    "value": "RECENT_POST",
                  },
                  "ref": null,
                  "rendered": "Postagens Recentes",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Postagens Antigas",
                    "value": "OLD_POST",
                  },
                  "ref": null,
                  "rendered": "Postagens Antigas",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nova Postagem",
              "className": "btn btn-outline-info btn-sm",
              "disabled": undefined,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Nova Postagem",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="navbar-brand"
            to="/"
>
            Udacity
</NavLink>,
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
>
            <ul
                        className="navbar-nav mr-auto"
            >
                        <li
                                    className="nav-item active"
                        >
                                    <NavLink
                                                activeClassName="active"
                                                ariaCurrent="true"
                                                className="nav-link"
                                                to="/"
                                    >
                                                Home 
                                                <span
                                                            className="sr-only"
                                                >
                                                            (current)
                                                </span>
                                    </NavLink>
                        </li>
            </ul>
</div>,
          <form
            className="form-inline "
>
            <div
                        className="mr-sm-2"
            >
                        <select
                                    className="form-control form-control-sm"
                                    disabled={undefined}
                                    id="sort"
                                    onChange={[Function]}
                                    value={undefined}
                        >
                                    <option
                                                value="SCORE_BIGGER"
                                    >
                                                Maior Pontuação
                                    </option>
                                    <option
                                                value="SCORE_SMALLER"
                                    >
                                                Menor Pontuação
                                    </option>
                                    <option
                                                value="RECENT_POST"
                                    >
                                                Postagens Recentes
                                    </option>
                                    <option
                                                value="OLD_POST"
                                    >
                                                Postagens Antigas
                                    </option>
                        </select>
            </div>
            <button
                        className="btn btn-outline-info btn-sm"
                        disabled={undefined}
                        onClick={[Function]}
                        type="button"
            >
                        Nova Postagem
            </button>
</form>,
        ],
        "className": "navbar navbar-expand-lg navbar-dark bg-dark justify-content-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Udacity",
            "className": "navbar-brand",
            "to": "/",
          },
          "ref": null,
          "rendered": "Udacity",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="navbar-nav mr-auto"
>
              <li
                            className="nav-item active"
              >
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          className="nav-link"
                                          to="/"
                            >
                                          Home 
                                          <span
                                                        className="sr-only"
                                          >
                                                        (current)
                                          </span>
                            </NavLink>
              </li>
</ul>,
            "className": "collapse navbar-collapse",
            "id": "navbarSupportedContent",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item active"
>
                  <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    className="nav-link"
                                    to="/"
                  >
                                    Home 
                                    <span
                                                      className="sr-only"
                                    >
                                                      (current)
                                    </span>
                  </NavLink>
</li>,
                Array [],
              ],
              "className": "navbar-nav mr-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-link"
                    to="/"
>
                    Home 
                    <span
                                        className="sr-only"
                    >
                                        (current)
                    </span>
</NavLink>,
                  "className": "nav-item active",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "ariaCurrent": "true",
                    "children": Array [
                      "Home ",
                      <span
                        className="sr-only"
>
                        (current)
</span>,
                    ],
                    "className": "nav-link",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Home ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "(current)",
                        "className": "sr-only",
                      },
                      "ref": null,
                      "rendered": "(current)",
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mr-sm-2"
>
                <select
                                className="form-control form-control-sm"
                                disabled={undefined}
                                id="sort"
                                onChange={[Function]}
                                value={undefined}
                >
                                <option
                                                value="SCORE_BIGGER"
                                >
                                                Maior Pontuação
                                </option>
                                <option
                                                value="SCORE_SMALLER"
                                >
                                                Menor Pontuação
                                </option>
                                <option
                                                value="RECENT_POST"
                                >
                                                Postagens Recentes
                                </option>
                                <option
                                                value="OLD_POST"
                                >
                                                Postagens Antigas
                                </option>
                </select>
</div>,
              <button
                className="btn btn-outline-info btn-sm"
                disabled={undefined}
                onClick={[Function]}
                type="button"
>
                Nova Postagem
</button>,
            ],
            "className": "form-inline ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  className="form-control form-control-sm"
                  disabled={undefined}
                  id="sort"
                  onChange={[Function]}
                  value={undefined}
>
                  <option
                                    value="SCORE_BIGGER"
                  >
                                    Maior Pontuação
                  </option>
                  <option
                                    value="SCORE_SMALLER"
                  >
                                    Menor Pontuação
                  </option>
                  <option
                                    value="RECENT_POST"
                  >
                                    Postagens Recentes
                  </option>
                  <option
                                    value="OLD_POST"
                  >
                                    Postagens Antigas
                  </option>
</select>,
                "className": "mr-sm-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="SCORE_BIGGER"
>
                      Maior Pontuação
</option>,
                    <option
                      value="SCORE_SMALLER"
>
                      Menor Pontuação
</option>,
                    <option
                      value="RECENT_POST"
>
                      Postagens Recentes
</option>,
                    <option
                      value="OLD_POST"
>
                      Postagens Antigas
</option>,
                  ],
                  "className": "form-control form-control-sm",
                  "disabled": undefined,
                  "id": "sort",
                  "onChange": [Function],
                  "value": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Maior Pontuação",
                      "value": "SCORE_BIGGER",
                    },
                    "ref": null,
                    "rendered": "Maior Pontuação",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Menor Pontuação",
                      "value": "SCORE_SMALLER",
                    },
                    "ref": null,
                    "rendered": "Menor Pontuação",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Postagens Recentes",
                      "value": "RECENT_POST",
                    },
                    "ref": null,
                    "rendered": "Postagens Recentes",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Postagens Antigas",
                      "value": "OLD_POST",
                    },
                    "ref": null,
                    "rendered": "Postagens Antigas",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Nova Postagem",
                "className": "btn btn-outline-info btn-sm",
                "disabled": undefined,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Nova Postagem",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
